berkeley-mobile/Map/MapViewController.swift

```diff
@@ -402,6 +402,7 @@ extension MapViewController: SearchResultsViewDelegate {
     // drop new pin and show detail view on search
     func choosePlacemark(_ placemark: MapPlacemark) {
         // remove last search pin
+        removeAnnotations(type: MapMarker.self)
```

Comment:
this leads to the bug where if all Cafe map markers are visible for example, dropping any pin/map marker using search will cause the Cafe map markers to disappear.
I like your solution where we track the single current pin/map marker dropped using search. then we can remove only that annotation instead of removing all annotations.